### PR TITLE
#1054: Set query cursor to 'None' when returned as empty string.

### PR DESCRIPTION
**Pull request by tseaver**
See #1054.

### CODE REVIEW

File: gcloud/datastore/query.py [hunk 1]
```diff
@@ -418,7 +418,10 @@ def next_page(self):
         #       for discussion.
         entity_pbs, cursor_as_bytes, more_results_enum = query_results[:3]
 
-        self._start_cursor = base64.b64encode(cursor_as_bytes)
+        if cursor_as_bytes == b'':
```
Comment (dhermes) [inline gcloud/datastore/query.py]: I was imagining we'd fix this "at the source" (in `Connection.run_query`), but even doing that would've required a branch here.

**Comment by dhermes** [comment]
LGTM but I'm still not sure if `Connect.run_query` should remain unchanged.

**Comment by dhermes** [comment]
@tseaver I was hoping we could discuss my concern before merging. Should I have held off the LGTM to make it clearer?

**Comment by tseaver** [comment]
Sorry -- I did take the `LGTM` as a green light.  Maybe re-open #1054 with your extra concern?

**Pull request merged**